src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -152325,5 +152719,149 @@ public static object Test(object a)
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOfTargetDelegate, "(out int v) =>").WithArguments("v", "lambda expression", "D<int>").WithLocation(2, 12)
+                );
+        }
+
+        [Fact, WorkItem(45571, "https://github.com/dotnet/roslyn/issues/45571")]
+        public void AnonymousFunctionConversion_Issue45571()
+        {
+            var source =
+@"#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+delegate void MyDel<T>([AllowNull] T t);
+
+class C
+{
+    static void F<T>(T t, MyDel<T> f)
+    {
+        f(default);
+    }
+    static void Main()
+    {
+        F(new object(), maybeNull => maybeNull.ToString());
+    }
+}
+";
+            var comp = CreateCompilation(new[] { source, AllowNullAttributeDefinition });
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(45571, "https://github.com/dotnet/roslyn/issues/45571")]
+        public void AnonymousFunctionConversion_Issue45571_2()
+        {
+            var source =
+@"#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+public delegate T Del<T>([DisallowNull] object? obj);
+
+public class C
+{
+    public T M<T>(Del<T> del)
+    {
+        return del(new object());
+    }
+
+    public void M1()
+    {
+        M(obj => obj).ToString();
+    }
+}
+";
+            // Missing lambda conversion warnings tracked by https://github.com/dotnet/roslyn/issues/56668
+            var comp = CreateCompilation(new[] { source, DisallowNullAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (15,9): warning CS8602: Dereference of a possibly null reference.
+                //         M(obj => obj).ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "M(obj => obj)").WithLocation(15, 9)
+                );
+        }
+
+        [Fact, WorkItem(35564, "https://github.com/dotnet/roslyn/issues/35564")]
+        public void AnonymousFunctionConversion_Issue35564()
+        {
+            var source =
+@"#nullable enable
+using System;
+using System.Collections.Generic;
+
+class Program
+{
+    static void Main()
+    {
+        Action<object> a1;
+        a1 = F1;
+        a1 = (object? o) => { };
+
+        Action<IEnumerable<object>> a2;
+        a2 = F2;
+        a2 = (IEnumerable<object?> e) => { };
+    }
+
+    static void F1(object? o) { }
+
+    static void F2(IEnumerable<object?> e) { }
+}
+";
+            // https://github.com/dotnet/roslyn/issues/35564: Consider relaxing and allow implicit conversions of nullability (as we do for method group conversions).
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (11,14): warning CS8622: Nullability of reference types in type of parameter 'o' of 'lambda expression' doesn't match the target delegate 'Action<object>' (possibly because of nullability attributes).
+                //         a1 = (object? o) => { }; // warning
```

Comment:
nit: it looks like // warning comments are present in this test baseline but not in the source code.